src/commons/error.rs

```diff
@@ -111,6 +111,45 @@ impl fmt::Display for RoaDeltaError {
     }
 }
 
+// ApiAuthError is *also* implemented as a separate enum,
```

Comment:
A few suggestions here:
  - rewrap this comment to 120 characters.
  - change "keep these errors as a separate container" to "keep these errors separate" as I find the reference to a container confusing.
  - it's untrue that "they all originate in interactions with the OIDC provider" as e.g. the MasterTokenAuthProvider and ConfigFileAuthProvider also both return the ApiInvalidCredentials error, and the ApiInsufficientRights error is returned from multiple places.